plugins/permission-node/src/ServerPermissionClient.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TokenManager, ServerTokenManager } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import {
+  AuthorizeRequestOptions,
+  DiscoveryApi,
+  PermissionClient,
+} from '@backstage/plugin-permission-common';
+
+/**
+ * A server side {@link @backstage/plugin-permission-common#PermissionClient}
+ * that allows all backend-to-backend requests.
+ * @public
+ */
+export class ServerPermissionClient extends PermissionClient {
```

Comment:
I think it would be best to use composition here instead, only calling out to the inner permission client when permissions are enabled, and skip the `shouldBypass` method

Reply:
Refactored as suggested! Used an interface as well, as suggested by @freben